### PR TITLE
Manage subscriptions and pages refactor

### PR DESCRIPTION
**Pull request by elbotho**
Supersedes #805

@Entkenntnis Sorry, this a bit much in one go,
It's the page refactoring and also a new feature.

http://localhost:3000/subscriptions/manage

### CODE REVIEW

**Comment by Entkenntnis** [comment]
Wow, that's a big one xD. I will take a look at it 👍

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -54,8 +55,12 @@ Router.events.on('routeChangeStart', () => {
 Router.events.on('routeChangeComplete', () => NProgress.done())
 Router.events.on('routeChangeError', () => NProgress.done())
 
-// eslint-disable-next-line import/no-default-export
-export default function App({ Component, pageProps }: AppProps) {
+interface CustomAppProps {
+  Component: NextComponentType<NextPageContext, any, InitialProps>
+  pageProps: InitialProps
+}
```
Comment (Entkenntnis) [inline src/pages/_app.tsx]: Using normal AppProps should suffice here. (InitialProps will be removed soon and replaced by page specific prop types)

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -54,8 +55,12 @@ Router.events.on('routeChangeStart', () => {
...
+
+export default function App({ Component, pageProps }: CustomAppProps) {
   return (
     <React.StrictMode>
```
Comment (Entkenntnis) [inline src/pages/_app.tsx:59]: Not necessary anymore

File: src/pages/spenden.tsx [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+// eslint-disable-next-line @typescript-eslint/require-await
+export const getStaticProps: GetStaticProps = async () => {
+  return {
+    props: {
+      pageData: {
+        kind: 'donation',
+      },
+    },
+  }
+}
```
Comment (Entkenntnis) [inline src/pages/spenden.tsx]: Not necessary

**Review by Entkenntnis — CHANGES_REQUESTED** [review]
Looks good! Some small leftovers from previous refactoring. I will merge after deployment, so we have time to test it on staging.

**Comment by elbotho** [comment]
@Entkenntnis Thank you for the review! Just added the changes.
Jup merging after Deploy is probably a good idea 👍 
Will base next PR on this, check?

**Pull request merged**